### ISSUE TITLE
Better error handling for invalid term entries

### ISSUE THREAD

**Issue by jmealo** [SlateFoundation/slate#68]
When the terms table is empty/invalid; we c/should handle the error by running the renesting operation automatically before failing with a friendly error "Invalid terms. Please contact your Slate administrator"

**Comment by themightychris** [comment]
I don't agree that renest should ever be run automatically, there is no situation where the renest would be necessary other than when a developer manually edits tables and it could be destructive to a table needing repair. There are three things we should focus on to improve term issues:

1) Have slate on setup initialize a current and next school year term
2) All tools should deal gracefully with there being no current term
3) All tools that can should use the `Term::getClosest()` call rather than `Term::getCurrent()` if possible
4) The term editing GUI should be hardened for production use

**Comment by jmealo** [comment]
@themightychris updated ticket to reflect work to be done
